core/src/test/scala/unit/kafka/server/ProduceRequestTest.scala

```diff
@@ -246,6 +248,18 @@ class ProduceRequestTest extends BaseRequestTest {
     assertEquals(Errors.UNSUPPORTED_COMPRESSION_TYPE, Errors.forCode(partitionProduceResponse2.errorCode))
   }
 
+  @Test
+  @Timeout(value = 3, unit = TimeUnit.SECONDS)
+  def testSendWithBatchSizeZeroAndNoRecordKey(): Unit = {
```

Comment:
I think this is a better place for the test: https://github.com/apache/kafka/blob/trunk/core/src/test/scala/integration/kafka/api/PlaintextProducerSendTest.scala#L50. Besides, there's already a test for batch size 0. All you need to add is to test with null key.

Reply:
Agreed, that's a much better place for it, thanks for the suggestion Luke. I've moved the test.